mage_ai/api/presenters/GitCustomBranchPresenter.py

```diff
@@ -21,6 +21,15 @@ def present(self, **kwargs):
 
         data_to_display = self.model
 
+        for key in [
+            'modified_files',
+            'staged_files',
+            'untracked_files',
+        ]:
+            arr = data_to_display.get(key)
+            if arr and type(arr) is list and len(arr) >= 1:
+                data_to_display[key] = arr[:100]
```

Comment:
This might not only be an issue in presenter.
I wonder whether the resource takes a long time to get the file list. If so, we'll limit the time in the resource.